### PR TITLE
qemu driver: Add option to configure drive_interface

### PR DESCRIPTION
**Pull request by bogue1979**
For QEMU the options `-drive file=image.img` and `-drive file=image.img,if=ide` are identical. 
This change will use the second variant as default and allow to define an image file using the virtio driver inside of the VM.
Using this driver should improve the disk IO a lot.

### CODE REVIEW

**Comment by bogue1979** [comment]
Ping

**Review by tgross — COMMENTED** [review]
Hi @bogue1979! This looks great. I think the only thing that comes to mind here is being sure about the default behavior for QEMU. Do you know if `ide` is platform-specific behavior and that QEMU isn't doing something "smart" based on the available interfaces on the platform? I want to make sure we're not introducing a breaking change, and that we're not boxing ourselves in for future updates we have on the roadmap for expanding QEMU support beyond x86_64 platform.

The [man page](https://linux.die.net/man/1/qemu-kvm) has this to say:
> if=interface
> This option defines on which type on interface the drive is connected. Available types are: ide, scsi, sd, mtd, floppy, pflash, virtio.

With nothing about the default.

**Comment by bogue1979** [comment]
> Hi @bogue1979! This looks great. I think the only thing that comes to mind here is being sure about the default behavior for QEMU. Do you know if `ide` is platform-specific behavior and that QEMU isn't doing something "smart" based on the available interfaces on the platform? I want to make sure we're not introducing a breaking change, and that we're not boxing ourselves in for future updates we have on the roadmap for expanding QEMU support beyond x86_64 platform.
> 
> The [man page](https://linux.die.net/man/1/qemu-kvm) has this to say:
> 
> > if=interface
> > This option defines on which type on interface the drive is connected. Available types are: ide, scsi, sd, mtd, floppy, pflash, virtio.
> 
> With nothing about the default.

Honestly I have no possibility to test another architecture. When I run a VM only with `file=image.img` on linux_x64, qemu implicitly attaches a virtual `ide` disk. On the other side giving the option to explicitly configure the interface, any non default behaviour can be overriden within the nomad job file. I guess if another architecture does not understand the given `ide` interface updating the documentation should be sufficient.

**Comment by bogue1979** [comment]
@tgross 
any news on this?

File: drivers/qemu/driver.go
Comment (tgross) [inline drivers/qemu/driver.go:353]: This isn't in a hot path, but no reason to allocate this whole list for each function call:

```suggestion
// hardcoded list of drive interfaces, Qemu currently supports
var allowedDriveInterfaces = []string{"ide", "scsi", "sd", "mtd", "floppy", "pflash", "virtio", "none"}

func isAllowedDriveInterface(driveInterface string) bool {
	for _, ai := range allowedDriveInterfaces {
```

**Review by tgross — CHANGES_REQUESTED** [review]
Sorry about the delayed follow-up @bogue1979. This looks pretty good.

You haven't made the PR editable by maintainers, so I'll also need you to add a changelog entry at `.changelog/11864.txt` with the following contents:

    ```release-note:improvement
    qemu: Added option to configure `drive_interface`
    ```

Once that's done we can get this merged and shipped in the upcoming Nomad 1.3.2

**Review by tgross — APPROVED** [review]
LGTM! Thanks for your patience on this one @bogue1979 !

**Comment by bogue1979** [comment]
Sorry I'm not really a developer and Git sometimes confuses me when it does not work as expected.

**Review by tgross — APPROVED** [review]

**Pull request merged**
